### DISCUSSION TITLE
7590AX - After the update (7.57) and the automatic restart of the box, the AVM-DSLD does not start correctly

### DISCUSSION THREAD

**Discussion by dl8obh** [Freetz-NG/freetz-ng#883] [Allgemein]
After the update and the automatic restart of the box, the AVM-DSLD does not start correctly. There is a temporary sync with the DSL line (with regular resyncs), the power LED lights up permanently, but an Internet connection (IP) is not established. The same problem also occurs when using an external modem.

https://www.ip-phone-forum.de/threads/nach-update-auf-fos-7-50-startet-der-avm-dsld-nicht-und-ein-root-login-per-telnet-ist-nicht-m%C3%B6glich.315112/

**Comment by fda77** [comment]
In den 2 letzten Posts steht dass es geht

**Comment by aus-sua** [comment]
The problem should have been resolved [since around the end of August 2023](https://github.com/Freetz-NG/freetz-ng/commits/master?after=652d8bb9b79fc4a3380e855b6e9e455fca4d950d+454&branch=master&qualified_name=refs%2Fheads%2Fmaster) (with the changes
to the wrapper for multid/dsld fixed)?
Which freetz-NG version are you using (but without logs and .config nobody to help!).

PS:
Before the wrapper changes and fixed, I used a 7490 (as the worst hack...) with:
`nohup sleep 8 && /sbin/multid && /sbin/dsld -C && /sbin/eventadd 1 "Uebelster Hack: multid + dsld per onlinechanged-offline" &`
in the freetz-NG--onlinechanged at:
offline*)
That worked for about 3 months without any problems... ;-)

---
Das Problem sollte doch [seit ca. Ende August 2023](https://github.com/Freetz-NG/freetz-ng/commits/master?after=652d8bb9b79fc4a3380e855b6e9e455fca4d950d+454&branch=master&qualified_name=refs%2Fheads%2Fmaster) mit den Änderungen
am wrapper für multid/dsld behoben sein?
Welche freetz-NG-Version benutzt Du (ohne logs und .config wird Dir aber hier keiner helfen können!)?

PS:
Ich hatte mich damals, vor den wrapper-Änderungen (quasi als übelster Hack), bei einer 7490 mit:
`nohup sleep 8 && /sbin/multid && /sbin/dsld -C && /sbin/eventadd 1 "Uebelster Hack: multid + dsld per onlinechanged-offline" &`
in der freetz-NG--onlinechanged bei
offline*)
beholfen.
Das funktionierte damit ca. 3 Monate lang ohne Probleme... ;-)

**Reply by fda77** [reply]
Ich glaube übrigens dass das Problem die Rechte von /tmp waren die tar änderts. Evtl hatte ich damit kein Problem da es den Benutzer den ich zum bauen nutze auch auf der Fritzbox gibt. Weshalb es "vorher" (wovor?) keine Probleme gab hab ich keine Idee. 
Aber wie gesagt ich hatte da nie Probleme

**Reply by aus-sua** [reply]
Ich bin damals mit meiner bescheidener Fehlersuche auch nicht weiter gekommen...
Ja, ich glaube auch, daß das tar für mehrere "Pseudo"-Fehler verantwortlich war...

**Comment by dl8obh** [comment]
Ich mach das auf die Schnelle auch mal auf Deutsch...

1. Mit einer 7490 gibt es diese Probleme NICHT !!!
2. Build-Umgebungen - in VirtualBox: 1.te Versuche mit [Freetz-Linux-2.2.1-Ubuntu 23.04. LTS (Lunar Lobster) 64-Bit], danach mit [Ubuntu 23.10 (Mantic Minotaur)]
3. freetz-ng 22722-e8fd874ba master 2023-10-29 -> 7590AX_07.57.all_freetz-ng-22722-e8fd874ba_20231030-104112.image
4. [.config_freetz-ng_22722-e8fd874ba_master_2023-10-29.txt](https://github.com/Freetz-NG/freetz-ng/files/13260375/default.config_freetz-ng_22722-e8fd874ba_master_2023-10-29.txt)
5. Die "Tricks" mit dem manuellen Stoppen und Starten des dsld habe ich dann beim Lesen des Threads auf [https://www.ip-phone-forum.de](https://www.ip-phone-forum.de/threads/nach-update-auf-fos-7-50-startet-der-avm-dsld-nicht-und-ein-root-login-per-telnet-ist-nicht-m%C3%B6glich.315112/) auch gelernt.
6. rc.custom: #
# /var/tmp/flash/mod/rc.custom
#
#
# DSLD Daemon (korrekt) starten
#
sleep 3
#
/sbin/dsld -s
#
sleep 3
#
/sbin/dsld -n -C
#
Habe ich aber noch nicht besonders ausgiebig getestet. Mir würde es wesentlich besser gefallen, wenn das so super wie VOR 7.50 laufen würde, ohne diese ganzen Klimmzüge !!! Ich bin BEGEISTERTER Freetz-Verwender !!! Damit mach ich auch routing ins/aus dem Amateurfunk-HAM-Net !!!
7. Wo finde ich die dementsprechenden Logfiles ???

**Reply by fda77** [reply]
Ich hab nur eine 7490 mit der ich testen könnte :) Du musst halt irgend wie herausfinden weshalb der Start fehlschlägt, nicht erfolgt oder der Daemon crasht. Vielleicht hast du auch irgend was noch in der rc.custom drin was das erst versusacht? oder in der onlinchanged viele sleep etc

**Reply by dl8obh** [reply]
Bis zu diesem Problem hatte ich GARNIX in'ner rc.custom. Was oder wo ist die "onlinchanged" ???

**Reply by fda77** [reply]
Ich werd jetzt sicher nicht 100e Möglichkeiten runterbeten und du sagst jedes mal "nö"

**Reply by dl8obh** [reply]
Doch mach doch mal, irgendwann ruft es bestimmt mal  B I N G O !!! Bis auf die "Extrawürste" in'ner .config, also zusätzliche Packages habe ich bisher NIX am "Standard-Freetz" geändert. Im Gegenteil: Ich hab' extra zum Testen ein Image gebaut, wo ich in 'nem "frischen" git clone nix weiter als den Box-Typ in'ner .config eingestellt habe, weiter nichts geändert, also STANDARD. Das Image tats auch nicht, daran kanns also auch nicht liegen, das kannste bei Deinen 100 Möglichkeiten also auch ausschliessen. Ebenfalls Änderungen an rc.custom oder online/offlinechanged, war nie notwendig. Mir fehlt mangels Erfahrung aber der Ansatz, wo ich gucken kann/muss...
"Gefühlt" oder "Gerochen" siehts für mich nach einer Racing-Codition aus: Der Prozess wird vermutlich zu früh gestrtet, sodass ein "Kommunikationspartner" noch gar nicht verhanden/verfügbar ist. "ps" zeigt mir den laufenden Prozess an, der sieht auch nicht "gecrasht" aus. Halt vermutlich mangels fehlender/noch nicht verfügbarer  Schnittstelle TAUBSTUMM ! Stoppe ich den Prozess und starte ihn dann manuell neu funktioniert es !!!

**Reply by fda77** [reply]
Ich hab keine Ahnung wo es klemmt sonst hätte ich es behoben.
Setzt doch mal alle avm+freetz einstellungen zurück und schau ob es dann geht. Oder für den Anfang nur Freetz
